python/GangaDirac/Lib/Backends/DiracBase.py

```diff
@@ -536,45 +541,59 @@ def _internal_job_finalisation(job, updated_dirac_status):
         logger = getLogger()
 
         if updated_dirac_status == 'completed':
+            start = time.time()
             # firstly update job to completing
             DiracBase._getStateTime(job, 'completing')
             if job.status in ['removed', 'killed']:
                 return
-            if (job.master and job.master.status in ['removed', 'killed']):
+            elif (job.master and job.master.status in ['removed', 'killed']):
                 return  # user changed it under us
 
             job.updateStatus('completing')
             if job.master:
                 job.master.updateMasterJobStatus()
 
-            import time
-            start = time.time()
             # contact dirac for information
-            job.backend.normCPUTime = execute('normCPUTime(%d)' % job.backend.id)
-            getSandboxResult = execute("getOutputSandbox(%d,'%s')" % (job.backend.id, job.getOutputWorkspace().getPath()))
-            file_info_dict = execute('getOutputDataInfo(%d)' % job.backend.id)
+            # The following single execute command is equivalent to the following few lines
+            #job.backend.normCPUTime = execute('normCPUTime(%d)' % job.backend.id)
+            #getSandboxResult = execute("getOutputSandbox(%d,'%s')" % (job.backend.id, job.getOutputWorkspace().getPath()))
+            #file_info_dict = execute('getOutputDataInfo(%d)' % job.backend.id)
```

Comment:
Frankly I find it easier to remember what is called here for reference sake when comparing to previous implementations but I'll drop it if it's controversial